types/README.md

```diff
@@ -0,0 +1,11 @@
+# Type declarations
+
+This folder contains custom type definitions for dependencies.
+
+https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html
+
+## Tests
+
+[check-dts](https://github.com/ai/check-dts) is used to test the type definitions.
+
+To run tests, run `yarn check-types`.
```

Comment:
Will these run on PRs?

Reply:
I didn't add them to PR tests here, thinking that the type declarations are isolated / unaffected by other changes and are unlikely to be frequently changed and responsibility for running them could be left up to whoever is making changes. I'm not opposed to adding them though.

Reply:
If the tests are fast to run, I don't see any reason not to run them on builds-- I would think it could catch issues introduced by library ugrades.

Reply:
Added in https://github.com/DataBiosphere/terra-ui/pull/3404/commits/35c10db2999daf58f5218415692dca96493614df.